### PR TITLE
Update multi-view prototype to 2023-08-30

### PR DESCRIPTION
**Pull request by loic-sharma**
/cc @dkwingsmt

Files that are worth reviewing:

```
lib/ui/hooks.dart
lib/ui/platform_dispatcher.dart
lib/ui/window/platform_configuration.cc
lib/ui/window/platform_configuration.h
runtime/runtime_controller.cc
shell/common/animator.cc
shell/common/engine.cc
shell/common/rasterizer.cc
shell/common/rasterizer.h
shell/common/shell.cc
shell/common/shell.h
shell/platform/windows/client_wrapper/flutter_engine.cc
runtime/runtime_controller.h
shell/common/engine.h
shell/platform/embedder/embedder.cc
```

### CODE REVIEW

File: lib/ui/hooks.dart [hunk 1]
```diff
@@ -44,14 +84,104 @@ void _updateDisplays(
...
 
-@pragma('vm:entry-point')
-void _sendViewConfigurations(List<int> viewIds) {
-  PlatformDispatcher.instance._onSentViewConfigurations(viewIds);
+// TODO(loicsharma): Should we remove this?
+// @pragma('vm:entry-point')
+// void _sendViewConfigurations(List<int> viewIds) {
+//   PlatformDispatcher.instance._onSentViewConfigurations(viewIds);
+// }
+
```
Comment (dkwingsmt) [inline lib/ui/hooks.dart]: Yes, we should remove this.
```suggestion
```

Comment (loic-sharma) [reply lib/ui/hooks.dart]: I will remove these, they're not necessary!

File: lib/ui/platform_dispatcher.dart [hunk 2]
```diff
@@ -274,31 +275,38 @@ class PlatformDispatcher {
...
     _invoke(onMetricsChanged, _onMetricsChangedZone);
   }
 
-  void _onSentViewConfigurations(List<int> viewIds) {
-    viewIds.forEach(_addView);
-  }
+  // TODO(loicsharma): Should we remove this?
+  // void _onSentViewConfigurations(List<int> viewIds) {
+  //   viewIds.forEach(_addView);
+  // }
```
Comment (dkwingsmt) [inline lib/ui/platform_dispatcher.dart]: Yep.
```suggestion
```

File: lib/ui/window/platform_configuration.cc [hunk 3]
```diff
@@ -85,46 +82,79 @@ void PlatformConfiguration::DidCreateIsolate() {
   report_timings_.Set(tonic::DartState::Current(),
                       Dart_GetField(library, tonic::ToDart("_reportTimings")));
 
-  library_.Set(tonic::DartState::Current(),
-               Dart_LookupLibrary(tonic::ToDart("dart:ui")));
+  // TODO(loicsharma): Can we remove this?
```
Comment (dkwingsmt) [inline lib/ui/window/platform_configuration.cc]: Both places in this file can be removed. They're already not in the main.

File: lib/ui/hooks.dart (hunk 1 above)
Comment (loic-sharma) [reply lib/ui/hooks.dart]: Removed

File: lib/ui/platform_dispatcher.dart (hunk 2 above)
Comment (loic-sharma) [reply lib/ui/platform_dispatcher.dart]: Removed

File: lib/ui/window/platform_configuration.cc (hunk 3 above)
Comment (loic-sharma) [reply lib/ui/window/platform_configuration.cc]: Removed

**Comment by dkwingsmt** [comment]
Yep, I've checked all files in your list and they look great!

**Pull request merged**